### PR TITLE
Fix of model_dims inside model

### PR DESCRIPTION
**Pull request by jidno**
There was bug when dims and model_dims were different which this fix should solve.

Also changed recon loss to be between decoder output and input (instead of input embedding).

Misc. change to get rid of mutable default argument, just to be safe. 

P.S.: Training the model looks good and converges.

### CODE REVIEW

**Comment by lucidrains** [comment]
@jidno oh already! thank you Theo for testing it out so soon! 🙏

**Comment by lucidrains** [comment]
@jidno in the limitations section of the paper, they said that convergence is sensitive to the learning rate

**Comment by jidno** [comment]
> @jidno in the limitations section of the paper, they said that convergence is sensitive to the learning rate

@lucidrains Just tested at three learning rates, 1e-3 which didn't converge, and at 1e-4 and 3e-4 which converged at similar loss values where 1e-4 got the better results.

**Comment by lucidrains** [comment]
@jidno what sort of data are you training on? is it something with a time component?

**Comment by jidno** [comment]
@lucidrains yep, 3D motion

**Comment by lucidrains** [comment]
@jidno nice! 🚀
